string object_type
int16 x_difference